openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -241,6 +244,15 @@ def from_component(cls, library_key, component):
         last_draft_created = draft.created if draft else None
         last_draft_created_by = draft.publishable_entity_version.created_by if draft else None
 
+        tags_count = 0
+        if load_tags_count:
+            # Build 'lib:' component usage_key from component slug and library_key
+            component_key = str(component).replace('xblock.v1:', f'{str(library_key)}:').replace('lib:', 'lb:')
+            components_tags = _get_library_component_tags_count(library_key)
+
+            if component_key in components_tags:
+                tags_count = components_tags[component_key]
```

Comment:
I think we can get the tags only for the block here:
```suggestion
        if load_tags_count:
            from openedx.core.djangoapps.content_tagging.api import get_object_tag_counts
            
            component_tags = get_object_tag_counts(str(component), count_implicit=True)
            
            if component_key in components_tags:
                tags_count = components_tags[component_key]
```